src/Oscoin/Crypto/Address.hs

```diff
@@ -0,0 +1,106 @@
+    , AddressPrefix
+    , AddressPayload
+
+    -- * Creating an Address
+    , fromPublicKey
+
+    -- * Parsing binary blobs back into addresses
+    , parseAddress
+
+    -- * Pretty-printing an Address
+    , renderAddress
+    , fmtAddress
+    ) where
+
+import           Oscoin.Prelude
+
+import qualified Oscoin.Configuration as Config
+import           Oscoin.Crypto.Address.Internal
+import           Oscoin.Crypto.Address.Serialisation as Serialisation
+import           Oscoin.Crypto.Hash
+import           Oscoin.Crypto.PubKey
+
+import qualified Codec.CBOR.ByteArray as CBOR
+import qualified Codec.Serialise as CBOR
+import           Data.ByteArray (ByteArrayAccess, convert)
+import           Data.ByteString.BaseN
+                 (decodeBase32z, encodeBase32z, encodedText)
+import           Data.Tagged
+import qualified Formatting as F
+
+{------------------------------------------------------------------------------
+    Creating an Address
+------------------------------------------------------------------------------}
+
+data AddressCreationError =
+    UnsupportedConfiguration Config.Network
+    deriving Show
+
+-- | Creates an 'Address' given a 'Network' configuration and a 'PublicKey'.
+fromPublicKey
+    :: forall c.
+       ( ByteArrayAccess (ShortHash c)
+       , Hashable c (PublicKey c)
+       )
+    => Config.Network
+    -- ^ The 'Network' the node is running on.
+    -> PublicKey c
+    -- ^ A 'PublicKey'.
+    -> Either AddressCreationError (Address c)
+fromPublicKey network pk = runExcept $
+    case configToAddressTypeTag network of
+      Nothing -> throwError $ UnsupportedConfiguration network
+      Just at -> do
+          let prefix = AddressPrefix
+                { addressFormat   = AddressFormat AddressFormatTagCBOR
+                , addressType     = AddressType at
+                , protocolVersion = ProtocolVersion ProtocolVersion_V1
+                }
+              payload = Tagged . CBOR.fromByteString . convert . shortHash @c $ pk
+          pure $ Address prefix (AddressPayload payload)
+
+-- | Parses a base32z-encoded binary block back into an 'Address'.
+parseAddress
+    :: ( CBOR.Serialise  (ShortHash c)
+       , ByteArrayAccess (ShortHash c)
+       )
+    => ByteString
+    -> Either Serialisation.DeserializeError (Address c)
+parseAddress base32zBlob =
+    case decodeBase32z base32zBlob of
+        Nothing -> Left $ Serialisation.InvalidBase32zEncoding base32zBlob
+        Just bs -> Serialisation.deserializeAddress bs
+
+{------------------------------------------------------------------------------
+    Using an Address
+------------------------------------------------------------------------------}
+
+-- | Renders an 'Address' into its textual form (i.e. a z-base-32-encoded
+-- blob).
+renderAddress
+    :: ( CBOR.Serialise (ShortHash c)
+       , ByteArrayAccess (ShortHash c)
+       )
+    => Address c
+    -> Text
+renderAddress = F.sformat fmtAddress
```

Comment:
Can we call this `showAddress` to keep with common vocabulary?

Reply:
We do have a bunch of "render" stuff, i.e. [renderEnvironment](https://github.com/oscoin/oscoin/blob/790c8cbe3c369a1875a9dcb29a0220d38e04feb7/src/Oscoin/Configuration.hs#L10) and [renderNetwork](https://github.com/oscoin/oscoin/blob/790c8cbe3c369a1875a9dcb29a0220d38e04feb7/src/Oscoin/Configuration.hs#L16). I think the "unspoken" rule I have been using is that `showBlah` is for things which either return a `String` or are close to the `Show` class (i.e. machine readable things), whereas `render` is for human.

I am flexible, though 😉

Reply:
Ah my bad, keep it as-is :+1: